cmd/influxd/launcher/print_config.go

```diff
@@ -0,0 +1,88 @@
+package launcher
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/influxdata/influxdb/v2/kit/cli"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v3"
+)
+
+func NewInfluxdPrintConfigCommand(v *viper.Viper, influxdOpts []cli.Opt) *cobra.Command {
+
+	var keyToPrint string
+	printOpts := make([]cli.Opt, len(influxdOpts)+1)
+
+	printOpts[0] = cli.Opt{
+		DestP: &keyToPrint,
+		Flag:  "key-name",
+		Desc:  "config key name; if set, only the resolved value of that key will be printed",
+	}
+	for i, opt := range influxdOpts {
+		printOpts[i+1] = cli.Opt{
+			DestP:  opt.DestP,
+			Flag:   opt.Flag,
+			Hidden: true,
+		}
+	}
+
+	cmd := &cobra.Command{
+		Use:   "print-config",
+		Short: "Print the full influxd config resolved from the current environment",
+		Long: `
+Print config (in YAML) that the influxd server would use if run with the current flags/env vars/config file.
```

Comment:
Where does the config file read happen in this code path (I'm sure it does, looking to be educated)

Reply:
`cli.NewCommand` (called [here](https://github.com/influxdata/influxdb/pull/20524/files#diff-7002d794d2d06b59d9096a9b955d58ccb645a3c1a9c8d66fb4eb9dd554a25da8R40)) is where config & env get loaded into the `Viper` instance. `cli.BindOptions` (called [here](https://github.com/influxdata/influxdb/pull/20524/files#diff-b37fd099a57923e8dbb2276b8dbcf3ced555f3ef55e95ca3ba5de35846b783f0R64)) is where the CLI flags get bound to local vars, with any config/env values injected as defaults.